### PR TITLE
Remove dual ROS 1+2 devcontainer, remove ROS Galactic from the support matrix

### PR DESCRIPTION
**Pull request by jhurliman**
### Public-Facing Changes

- ROS Galactic support has been dropped from the support matrix

### Description

In anticipation of adding ROS 2-specific functionality that relies on ROS Humble or later, ROS Galactic has been removed from the support matrix. The dual ROS 1+2 devcontainer has been replaced with a ROS 2 Humble devcontainer that matches what is used in CI.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ros_distribution: [melodic, noetic, galactic, humble, iron, rolling]
+        ros_distribution: [melodic, noetic, humble, iron, rolling]
```
Comment (achim-k) [inline .github/workflows/ci.yml:24]: maybe we should leave that? Technically people could still manually build for `galactic`

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -32,7 +31,5 @@
   },
   "capAdd": [
     "SYS_PTRACE"
-  ],
-  // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
-  "remoteUser": "vscode"
```
Comment (achim-k) [inline .devcontainer/devcontainer.json:37]: What's the impact of removing this? I think files will be owned by root which messes with permissions on the host system, not sure anymore.

**Review by achim-k — COMMENTED** [review]
> In anticipation of adding ROS 2-specific functionality that relies on ROS Humble or later, ROS Galactic has been removed from the support matrix.

Does that mean that your anticipated changes will prevent users from manually building foxglove_bridge for galactic?

**Review by achim-k — APPROVED** [review]

**Comment by jhurliman** [comment]
> > In anticipation of adding ROS 2-specific functionality that relies on ROS Humble or later, ROS Galactic has been removed from the support matrix.
> 
> Does that mean that your anticipated changes will prevent users from manually building foxglove_bridge for galactic?

Yes, that's correct. Once `ros2_babel_fish` code is vendored into this repo, anyone trying to build `ros-foxglove-bridge` against a ROS 2 release before Humble will see compilation errors about trying to call methods that don't exist.

**Pull request merged**